proxy/src/connection.rs

```diff
@@ -165,19 +215,25 @@ impl BoundPort {
                     set_nodelay_or_warn(&socket);
 
                     let tls = match &tls {
-                        Conditional::Some(tls) => match &*tls.config.borrow() {
-                            Conditional::Some(config) =>
-                                Conditional::Some(tls::ConnectionConfig {
-                                    identity: tls.identity.clone(),
-                                    config: config.clone(),
-                                }),
-                            Conditional::None(r) => Conditional::None(*r),
-                        },
+                        Conditional::Some(TlsAccept { sensor, config: conn_config }) =>
+                            match &*conn_config.config.borrow() {
+                                Conditional::Some(config) => {
+                                    let config = tls::ConnectionConfig {
+                                        identity: conn_config.identity.clone(),
+                                        config: config.clone(),
+                                    };
+                                    Conditional::Some((sensor, config))
```

Comment:
Let's order things `(config, sensor)` to be consistent with the rest of the stuff. It seems like if we can use a tuple here then we don't need `TlsConnect` and `TlsAccept`; conversely, if we're going to have those then why not use them here?